api/v1beta1/nutanixcluster_types.go

```diff
@@ -107,6 +129,8 @@ type NutanixCluster struct {
 }
 
 // NutanixFailureDomain configures failure domain information for Nutanix.
+//
+// Deprecated: This type is replaced by the NutanixFailureDomainV2 CRD type and will be removed in the next apiVersion.
 type NutanixFailureDomain struct {
```

Comment:
Rename the existing inline type used under NutanixClusterSpec.failureDomains from NutanixFailureDomain to NutanixFailureDomainConfig since Kubernetes CRDs are purely “JSON on-the-wire” and driven by field tags (not Go type names), renaming your Go struct has zero impact on the existing schema or stored objects: as long as you keep the same json:"failureDomains" tag. Kubebuilder and controller-gen look solely at those JSON tags to build the OpenAPI schema in your CRD.